hazelcast/src/test/java/com/hazelcast/client/map/ClientMapIteratorTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.map;
+
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.map.AbstractMapIteratorTest;
+import com.hazelcast.map.IMap;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Iterator;
+import java.util.Map;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ClientMapIteratorTest extends AbstractMapIteratorTest {
+
+    @Override
+    public void setup() {
+        factory = new TestHazelcastFactory();
+        factory.newHazelcastInstance(getConfig());
+        instance = factory.newHazelcastClient();
+    }
+
+    @Override
+    protected <K, V> Iterator<Map.Entry<K, V>> getIterator(IMap<K, V> map) {
+        return map.iterator();
```

Comment:
So this would result in code replication. Because `AbstractMapIterableTest` extends `AbstractMapIteratorTest` since all iterable tests use iterator tests. If we change `AbstractMapIteratorTest` then we would need to copy it's contents to `AbstractMapIterableTest`